lib/elixir_internal_certification_worker/google.ex

```diff
@@ -0,0 +1,42 @@
+defmodule ElixirInternalCertificationWorker.Google do
+  use Oban.Worker, max_attempts: 4
+
+  import Ecto.Query, warn: false
+
+  alias ElixirInternalCertification.Fetcher.Google, as: GoogleFetcher
+  alias ElixirInternalCertification.Keyword.{KeywordLookups, Keywords}
+  alias ElixirInternalCertification.Keyword.Schemas.Keyword
+  alias ElixirInternalCertification.Parser.Google, as: GoogleParser
+
+  def perform(%Oban.Job{args: %{"keyword_id" => keyword_id}, attempt: 4} = _oban_job) do
+    keyword = Keywords.get_keyword!(keyword_id)
+    Keywords.update_status(keyword, :failed)
+    {:error, "Failed to look up the keyword ID: #{keyword_id}"}
+  end
+
+  def perform(%Oban.Job{args: %{"keyword_id" => keyword_id}} = _oban_job) do
+    keyword = Keywords.get_keyword!(keyword_id)
+    result = execute(keyword)
+
+    case result do
+      {:ok, _} ->
+        Keywords.update_status(keyword, :completed)
+        result
+
+      error ->
+        error
+    end
+  end
+
+  defp execute(%Keyword{} = keyword) do
+    %Keyword{id: keyword_id, title: keyword_title} = keyword
```

Comment:
```suggestion
  defp execute(%Keyword{id: keyword_id, title: keyword_title} = _keyword) do
```
As you are already matching on the Keyword struct in the argument it doesn't make much sense to then extract it in the body when you can do it at the same time.

I included `_keyword` at the end as you seem to like doing it when the value is not used. Not sure if any convention to include it or not when we don't use it

Reply:
Yeah I don't think there's convention to not do this so I am happy with it